docs/hub/webhooks-guide-metadata-review.md

```diff
@@ -0,0 +1,207 @@
+):
+    # [...]
+    return report
+```
+
+
+## How to post the review automatically?
+
+We now have a markdown formatted metadata review report. We'll use the `huggingface_hub` library to post this review. We define a function that takes back the webhook data received from the Hub, parses the data, and creates the metadata report. Depending on whether a report has previously been created, the function creates a new report or posts a new issue to an existing metadata review thread.
+
+```python
+def create_or_update_report(data):
+    if parsed_post := parse_webhook_post(data):
+        repo_type, repo_name = parsed_post
+    else:
+        return Response("Unable to parse webhook data", status_code=400)
+    # [...]
+    return True
+```
+
+## Creating a webook to respond to changes on the Hub
+
+We've now got the core functionality for creating a metadata review report for a model or dataset. The next step is to use webhooks to respond to changes automatically. 
+
+## Create a webook in your user profile
+
+First, create your webhook by going to https://huggingface.co/settings/webhooks.
+
+- Input a few target repositories that your webhook will listen to (you will likely want to limit this to your own repositories or the repositories of the organization you belong to).
+- Input a secret to make your webhook more secure (if you don't know what to choose for this, you may want to use a [password generator](https://1password.com/password-generator/) to generate a sufficiently long random string for your secret).
+- We can pass a dummy URL for the `Webhook URL` parameter for now. 
+
+Your webhook will look like this:
+
+TODO add screenshot
+
+## Create a new "Bot" user profile
+
+This guide creates a separate user account that will post the metadata reviews. 
+
+TODO bot profile screenshot
+
+<Tip>
+	When creating a bot that will interact with other users on the Hub, we ask that you clearly label the account as a "Bot" (see profile screenshot).
+</Tip>
+
+
+## Create a webook listener
+
+We now need some way of listening to webhook events. There are many possible tools you can use to listen to webhook events. Many existing services, such as [Zapier](https://zapier.com/) and [IFTTT](https://ifttt.com), can use webhooks to trigger actions (for example, they could post a tweet every time a model is updated). In this case, we'll implement our webhook listener using  [FastAPI](https://fastapi.tiangolo.com/). 
+
+[FastAPI](https://fastapi.tiangolo.com/) is a Python web framework. We'll use FastAPI to create a webhook listener. In particular, we need to implement a route that accepts `POST` requests on `/webhook`.
+
+
+```python
+from fastapi import FastAPI
+
+@app.post("/webhook")
+async def webhook(request: Request):
+    if request.method == "POST":
+        if request.headers.get("X-Webhook-Secret") != KEY:
+            return Response("Invalid secret", status_code=401)
+        payload = await request.body()
+        data = json.loads(payload)
```

Comment:
maybe there a built-in method (like await request.json maybe) to do this in just one line? (not sure)